src/gui/editors/AutomationEditor.cpp

```diff
@@ -134,13 +134,10 @@ AutomationEditor::AutomationEditor() :
 
 	// add time-line
 	m_timeLine = new TimeLineWidget(VALUES_WIDTH, 0, m_ppb,
-		Engine::getSong()->getPlayPos(Song::PlayMode::AutomationClip),
 		Engine::getSong()->getTimeline(Song::PlayMode::AutomationClip),
 		m_currentPosition, Song::PlayMode::AutomationClip, this
 	);
-	connect(this, &AutomationEditor::positionChanged, m_timeLine, &TimeLineWidget::updatePosition);
-	connect( m_timeLine, SIGNAL( positionChanged( const lmms::TimePos& ) ),
-			this, SLOT( updatePosition( const lmms::TimePos& ) ) );
+	connect(m_timeLine->model(), &Timeline::positionChanged, this, &AutomationEditor::updatePosition, Qt::QueuedConnection);
```

Comment:
I don't know how safe it is to remove stuff like `connect(this, &AutomationEditor::positionChanged, m_timeLine, &TimeLineWidget::updatePosition);` Anything that is still emitting the `positionChanged` signal in `AutomationEditor` (like `AutomationEditor::horScrolled`) will be broken. This is a similar situation I have encountered in your PR with classes like `PianoRoll`, `SongEditor`, etc. It looks like these two signals are separate, so maybe you should keep that connection instead of removing it?

Reply:
Whooops, I don't think I meant to remove that first `connect`.